### DISCUSSION TITLE
Profiling 101 - XX’s Blog

### DISCUSSION THREAD

**Discussion by giscus** [xxchan/xxchan.github.io#38] [Announcements]
# Profiling 101 - XX’s Blog

以前也浅尝辄止地试图 profile 过，但是被一大堆概念和工具搞的头昏脑涨，全是问题：    什么是 profiling，这个词什么意思？   CPU profiling 和 memory profiling 有什么区别？为什么都是火焰图？   Instrument 又是什么意思？   Golang 的 pprof package 和 pprof tool 有什么区别？   Mac 的 XCode Instruments 可以比较傻瓜式地 profile，但是没有 flamegraph，怎么办？

https://xxchan.github.io/cs/2023/02/08/profiling-101.html

**Comment by mapleFU** [comment]
个人经验，不一定对，一般排查性能问题是首先要了解这块代码，然后：
1. 看瓶颈在哪，这个瓶颈大小是否符合预期，从采样的从多到少来排查。
2. 非瓶颈的一些占用是否符合预期。这个可能有点生产，有的时候会在线上看到自己觉得不该出现的栈...有的时候甚至能根据这个找到 bug
当然也有可能自己某个写法就有问题，那就不太是火焰图帮的上忙的事情了（

**Comment by boholder** [comment]
"插桩"是stub吧，本来是用来在单测中mock依赖用的，提供一个假的底层。profiling根据工具不同，有时候确实跟MITM攻击一样给程序提供个插了监控逻辑的底层。。

**Comment by Endle** [comment]
请问博主是否有打算，写一篇在具体项目中的用火焰图调优的记录呢？

**Reply by xxchan** [reply]
如果有这种宝贵经历的话那估计我肯定会写的🤣
